### PR TITLE
Compact Block Tweaks (rebase of #8235)

### PR DESCRIPTION
**Pull request by sipa**
This is a rebase of #8235, with unit tests from #8418 adapted, and some comments addressed.

### CODE REVIEW

File: src/main.h [hunk 1]
```diff
@@ -88,6 +88,9 @@ static const unsigned int BLOCK_STALLING_TIMEOUT = 2;
 /** Number of headers sent in one getheaders result. We rely on the assumption that if a peer sends
  *  less than this number, we reached its tip. Changing this value is a protocol upgrade. */
 static const unsigned int MAX_HEADERS_RESULTS = 2000;
+/** Maximum age (in blocks) of blocks we're willing to serve as compact blocks to peers
+ *  when requested. For older blocks, a regular BLOCK response will be sent. */
+static const int MAX_COMPACT_BLOCK_DEPTH = 5;
```
Comment (instagibbs) [inline src/main.h]: It's been too long since I've looked at this code, but this sure seems close to `MAX_CMPCTBLOCK_DEPTH`

Comment (sipa) [inline src/main.h]: Where do you see MAX_CMPCTBLOCK_DEPTH?

Comment (laanwj) [inline src/main.h]: There's a MAX_CMPCTBLOCK_DEPTH in the `p2p-compactblocks.py` test, but I'd say it's not a concern here.

Comment (instagibbs) [inline src/main.h]: ah right I noticed that then forgot when looking this morning. Forgive me.

Comment (sipa) [inline src/main.h]: Oh, right. I'll make the constant names correspond.

**Comment by sipa** [comment]
Updated with exactly corresponding constant names between the tests and the implementation.

**Comment by gmaxwell** [comment]
Needs rebase

**Comment by sipa** [comment]
Rebased.

**Comment by gmaxwell** [comment]
ACK

**Comment by TheBlueMatt** [comment]
At the risk of holding things up, you might want https://github.com/TheBlueMatt/bitcoin/commit/654f83f66cc31cbde9b3f14b1c30bdfa8821fd31 as well.

**Comment by TheBlueMatt** [comment]
Actually, I'll do that separately, utACK on the contents here.

**Comment by TheBlueMatt** [comment]
Version rebased on #8393 with the cs_main fix on top (I'll pr that separately if you dont want to take it here)

**Comment by laanwj** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/8637/commits/1bcdb3dd9b9388700c42a099f025371199d3494e

**Comment by TheBlueMatt** [comment]
Needs rebase (see rebased version at https://github.com/TheBlueMatt/bitcoin/commits/cmpctblock).

**Comment by instagibbs** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/8637/commits/7e2f31e5ddd294a7783db025f51fd7d9f5d93e4d

**Comment by TheBlueMatt** [comment]
utACK 3ac6de0a045cc9b2047ceb19af970e7ffbf905fa

**Comment by btcdrak** [comment]
utACK 3ac6de0a045cc9b2047ceb19af970e7ffbf905fa

**Pull request merged**